lib/federation-utils.js

```diff
@@ -23,7 +26,51 @@ var compareFederateKeys = (publicKeysA, publicKeysB) => {
   return comparison;
 };
 
+const getNewFederationPublicKeysFromNewFederationConfig = (newFederationConfig) => {
+  const newFederationPublicKeys = newFederationConfig.members.map(member => member.publicKeys);
+  newFederationPublicKeys.sort((keyA, keyB) => keyA.btc.localeCompare(keyB.btc));
+  return newFederationPublicKeys;
+};
+
+const stopPreviousFederators = async (newFederationConfig) => {
+
+  // TODO: right now there is only the genesis federation and a second federation. So the previous is the genesis federation.
+  // When more are added, we need to change this logic to dynamically select the previous federation.
+  // I tried with a `switch` statement ready for future changes but sonar complains about it.
+  const previousFederationId = 'genesis-federation';
+
+  Runners.fedRunners.forEach((fedRunner) => {
+      if(fedRunner.options.federationId === previousFederationId) {
+          fedRunner.stop();
+      }
+  });
+
+  Runners.fedRunners = Runners.fedRunners.filter(fedRunner => fedRunner.options.federationId !== previousFederationId);
+  Runners.hosts.federates = Runners.hosts.federates.filter(fedHostInfo => fedHostInfo.federationId !== previousFederationId);
+  Runners.hosts.federate = Runners.hosts.federates[0];     
+
+};
+
+const startNewFederationNodes = async (newFederationConfig) => {
+  for(let i = 0; i < newFederationConfig.length; i++) {
+      const fedConfig = newFederationConfig[i];
+      await federateStarter.startFederate(i + 1, fedConfig);
+  }
+};
+
+const fundNewFederators = async (rskTxHelper, newFederationConfig) => {
+  for(const fedConfig of newFederationConfig) {
+      const fedRskCompressedPublicKey = fedConfig.publicKeys.rsk;
+      const fedRskAddress = rskUtils.getAddressFromUncompressedPublicKey(rskUtils.uncompressPublicKey(fedRskCompressedPublicKey));
```

Comment:
```suggestion
      const newFederatorRskCompressedPublicKey = fedConfig.publicKeys.rsk;
      const newFederatorRskAddress = rskUtils.getAddressFromUncompressedPublicKey(rskUtils.uncompressPublicKey(fedRskCompressedPublicKey));
```

no big deal, just find it more accurate